src/index.re

```diff
@@ -304,58 +304,49 @@ let drawInventory = (inventory, spriteData, ~dragging, ~hovered, ~time, env) =>
   );
 };
 
-let getUndoRect = env => {
-  let height = float_of_int(Env.height(env));
-  let backgroundY = height -. toolbarHeight;
-  let size = tileSizef *. 1.4;
-  let delta = btnMargin /. 2.;
-  Rect.fromPoints(
-    Point.create(
-      btnMargin +. size +. btnMargin +. delta,
-      btnMargin +. backgroundY -. delta,
-    ),
-    Point.create(size, size),
-  );
-};
-
-let getBackRect = env => {
-  let height = float_of_int(Env.height(env));
-  let backgroundY = height -. toolbarHeight;
-  let size = tileSizef *. 1.4;
-  let delta = btnMargin /. 2.;
-  Rect.fromPoints(
-    Point.create(btnMargin +. delta, btnMargin +. backgroundY -. delta),
-    Point.create(size, size),
-  );
-};
-
-let getPlayRect = env => {
-  let height = float_of_int(Env.height(env));
-  let backgroundY = height -. toolbarHeight;
-  let size = tileSizef *. 1.4;
-  let delta = btnMargin /. 2.;
-  Rect.fromPoints(
-    Point.create(
-      btnMargin +. size +. btnMargin +. size +. btnMargin +. delta,
-      btnMargin +. backgroundY -. delta,
-    ),
-    Point.create(size, size),
-  );
-};
+let useButton = (loc, rect, assetName, env) => {
+  let (pressed, setPressed) = Hooks.useState(__LOC__, ~id=loc, false);
```

Comment:
Did you mean to use `loc` here instead of `__LOC__`?

Reply:
no it uses both

Reply:
Oh I'm dumb!